apis/filesystem/src/test/java/org/jclouds/filesystem/strategy/internal/FilesystemStorageStrategyImplTest.java

```diff
@@ -686,9 +686,21 @@ public void testPutIncorrectContentLength() throws Exception {
       }
    }
 
-   @Test
-   public void testDeletingInvalidPathFileEndsNormally() {
+   @DataProvider
+   public Object[][] getInvalidPathBlobKey() {
       String invalidPathBlobKey = "A<!:!@#$%^&*?]8 /\0";
+      // the JDK can't handle '/' in Windows file paths
```

Comment:
`FilesystemStorageStrategyImpl.normalize` should handle this translation.  I wonder why it does not?

Reply:
Aaaaah...so what seems to be happening here is that removing `\` means that `parentPath` [here](https://github.com/jclouds/jclouds/blob/master/apis/filesystem/src/main/java/org/jclouds/filesystem/strategy/internal/FilesystemStorageStrategyImpl.java#L846) is then empty, so the code that throws the error (`directory.toPath()`, a few lines down) is never actually called.

@gaul Do you recall what the intention is here? Is `invalidPathBlobKey` supposed to represent a blob key **without** path information, or is it supposed to represent a path, too?